ARCs/arc-0018.md

```diff
@@ -0,0 +1,348 @@
+            "args":[
+                {"name":"royalty_asset", "type":"asset"}, 
+                {"name":"owner_acct", "type":"account"},
+                {"name":"buyer_acct", "type":"account"},
+                {"name":"royalty_receiver", "type":"account"},
+                {"name":"royalty_asset_amount", "type":"uint64"},
+                {"name":"payment", "type":"txn"},
+                {"name":"payment_asset", "type":"asset"}
+            ],
+            "returns":{"type":"void"}
+        }
+    ],
+    "locals": [ 
+        {
+            "name": "{uint64 asset id as big endian bytes}",
+            "desc": "The ID of the asset that has been offered as key, authorizing address as value",
+            "type": "(address,uint64)"
+        }  
+    ],
+    "globals": [ 
+        {
+            "name": "0000000000000000",
+            "desc": "The default royalty policy for any ASA unless its ASA ID is specified",
+            "type": "(address,uint64,uint64,uint64,uint64)"
+        },
+        {
+            "name": "{uint64 asset id as big endian bytes}",
+            "desc": "Royalty policy for asset ID in key ",
+            "type": "(address,uint64,uint64,uint64,uint64)"
+        }  
+    ]
+}
+```
+
+### Metadata
+
+The metadata associated to an asset MAY conform to [ARC-3](./arc-0003.md) or [ARC-69](./arc-0069.md) with an additional field in the `attributes` section specifying the Application ID for the Royalty Enforcer Contract. 
+
+```js
+//...
+"attributes":{
+    "arc18":{
+        "application-id":123,
+    }
+    //...
+}
+//...
+```
+
+## Rationale
+
+The motivation behind defining a Royalty Enforcement specification is the need to guarantee a portion of a payment is received by select royalty collector on sale of an asset. Current royalty implementations are either platform specific or are only adhered to when an honest seller complies with it, allowing for the exchange of an asset without necessarily paying the royalties.
+
+The use of a smart contract as a clawback address is a guaranteed way to know an asset transfer is only ever made when certain conditions are met, or made in conjunction with additional transactions. The Royalty Enforcer is responsible for the calculations required in dividing up and dispensing the payments to the respective parties. The present specification does not impose any restriction on the Royalty Collector distribution logic (if any), which could be achieved through a Multi Signature account, a Smart Signature or even through another Smart Contract.
+
+On Ethereum the EIP-2981 standard allows for ERC-721 and ERC-1155 interfaces to signal a royalty amount to be paid, however this is not enforced and requires marketplaces to implement and adhere to it.
+
+## Backwards Compatibility
+
+Existing ASAs with unset freeze and clawback roles will be incompatible with this specification.
```

Comment:
```suggestion
Existing ASAs with unset clawback address or unset manager address (in case the clawback address is not the application account of a smart contract that is updatable - which is most likely the case) will be incompatible with this specification.
```